rest-service/manager_rest/test/mocks.py

```diff
@@ -117,6 +103,23 @@ def _do_request(self, requests_method, request_url, body, params, headers,
         elif 'put' in requests_method.__name__:
             if isinstance(body, types.GeneratorType):
                 body = b''.join(body)
+            if hasattr(body, 'fields'):
+                # FlaskClient wants dicts, not MultipartEncoders or the like
+                body = body.fields
+                for entry in body:
+                    if (
+                        isinstance(body[entry], tuple)
+                        and body[entry][0] == 'filename'
+                    ):
+                        body[entry] = FileStorage(
+                            stream=open(body[entry][1].name, 'rb'))
+
+                # Remove application/json header to not override multipart
+                for header in headers:
+                    if headers[header] == 'application/json':
+                        break
+                headers.pop(header)
```

Comment:
this will always remove the last header. Write this differently, eg. using a dictcomp

Reply:
It won't, it'll remove the one that has the value of application/json- but yes, dictcomp would be better.

Reply:
and if there isn't one that has the value of application/json, then it'll remove an unrelated header that just happens to iterate last

Reply:
Also, that header's name is always content-type

Reply:
Ah, good point, yes- anyway, dictcomp version incoming.

Reply:
It's actually Content-type for the application/json. The one that gets added by the multipartencoder is Content-Type.